src/linuxforhealth/csvtofhir/fhirrs/basic.py

```diff
@@ -0,0 +1,163 @@
+    
+    # Create patient-tokens coding   
+    # "code": {
+    #    "coding": [
+    #         {
+    #             "code": "patient-tokens",
+    #             "display": "Patient Tokens",
+    #             "system": "http://ibm.com/fhir/cdm/CodeSystem/wh-basic-resource-type"
+    #         }
+    #     ],
+    #     "text": "Patient Tokens"
+    # }
+    code_cc = fhir_utils.get_codeable_concept(
+        "http://ibm.com/fhir/cdm/CodeSystem/wh-basic-resource-type",
+        "patient-tokens",
+        "Patient Tokens",
+        "Patient Tokens"
+    )
+      
+    # loop through token list and create token identifiers
+    identifierList = []
+    if incoming_data.tokenList:
+        for token in incoming_data.tokenList:
+            # tokenList list items are in the form of 
+            # ["token_name^token_value^system", "token_name^token_value^system", token_name^token_value^system"]
+            tokens = token.split("^")
+            token_name = tokens[0]
+            token_value = tokens[1]
+            token_system = tokens[2]
+            token_id = incoming_data.baseSystem + "." + token_name
+            
+            identifier = _build_token_identifier(token_id, token_value, token_system)
+            identifierList.append(identifier)
+            
+    if incoming_data.otherIdentifierList:
+        for other in incoming_data.otherIdentifierList:
+            # otherIdentifierList list items are in the form of 
+            # ["name^value^system", "name^value^system", name^value^system"]
+            parts = other.split("^")
+            name = parts[0]
+            value = parts[1]
+            system = parts[2]
+            id = incoming_data.baseSystem + "." + name
+            
+            identifier = _build_other_identifier(id, name, value, system)
+            identifierList.append(identifier)
+                  
+    # Use local copy of resource_meta so changes do not affect other resources
+    # Passing in none for the source record id because it is not applicable here
+    resource_meta_copy = fhir_utils.add_source_record_id_return_meta_copy(
+        None, resource_meta)
+    
+    # create basic fhir resource with an id if patientInternalIdentifier is present
+    fhir_resource: Basic = Basic.construct(
+        id=resource_id,
+        code=code_cc,
+        identifier=identifierList,
+        meta=resource_meta_copy
+    )
+    
+    # Add patient subject link if the patientInternalIdentifier is present
+    if incoming_data.patientInternalIdentifier != None and incoming_data.patientInternalIdentifier != "":
+        patient_reference = fhir_utils.get_resource_reference_from_str(
+            "Patient",
+            incoming_data.patientInternalIdentifier
+        )
+        fhir_resource.subject = patient_reference
+    
+    # Add created date if it's present
+    if incoming_data.created_date != None and incoming_data.created_date  != "":
+        time_part_index = incoming_data.created_date.index(" ")
```

Comment:
Does this work if there is no time (no space)?

Reply:
Let me add some code that wil handle just the date.

Reply:
Changed it to:
 # remove the time part of the date time if it exists
        time_part_index = incoming_data.created_date.index(" ")
        # if there is no time then use the end of the string.
        if time_part_index == -1:
            time_part_index = len(incoming_data.created_date)
        created_date = incoming_data.created_date[0:time_part_index]